### PR TITLE
Remove global mocks by adding a global flag

### PR DESCRIPTION
**Pull request by gaearon**
This is necessary for running a subset of tests against flat bundles.
I added a global unstable flag to avoid exposing yet another injection for error reporting.

Not ideal, but it works and can serve as an escape hatch to https://github.com/facebook/react/issues/11098.

### CODE REVIEW

**Comment by gaearon** [comment]
Development:

```
Test Suites: 32 failed, 87 passed, 119 total
Tests:       1202 failed, 26 skipped, 1192 passed, 2420 total
```

Production:

```
Test Suites: 59 failed, 60 passed, 119 total
Tests:       1366 failed, 26 skipped, 1029 passed, 2421 total
```

File: packages/react-reconciler/src/ReactFiberErrorLogger.js [hunk 1]
```diff
@@ -11,7 +11,21 @@ import type {CapturedError} from './ReactFiberScheduler';
...
+  // This is a somewhat hacky way to disable error logging.
+  // Can be handy in tests.
+  // We can revisit this when we offer a more comprehensive
+  // solution for overriding warning behavior.
+  if (
+    typeof __REACT_UNSTABLE_SUPPRESS_ERROR_LOGGING__ !== 'undefined' &&
+    __REACT_UNSTABLE_SUPPRESS_ERROR_LOGGING__ === true
+  ) {
+    return false;
+  }
```
Comment (bvaughn) [inline packages/react-reconciler/src/ReactFiberErrorLogger.js]: 🙃 
```js
return (
  typeof __REACT_UNSTABLE_SUPPRESS_ERROR_LOGGING__ !== 'undefined' &&
  __REACT_UNSTABLE_SUPPRESS_ERROR_LOGGING__ === true
);
```

**Comment by gaearon** [comment]
I mean... I could also look into some magic to suppress these warnings at the test site. For example by adding `toThrowAndWarnInDev`. Or even overriding the default `toThrow` behavior to ignore last warning if its message matches.

Adding this magic will probably make tests more confusing if we ever want to change logging behavior. But the upside is all the hacks are isolated to our tests, and won't affect Prepack etc.

**Comment by gaearon** [comment]
I like this more: https://github.com/facebook/react/pull/11636

**Comment by gaearon** [comment]
Went with https://github.com/facebook/react/pull/11636#merged-event

**Pull request closed without merging**